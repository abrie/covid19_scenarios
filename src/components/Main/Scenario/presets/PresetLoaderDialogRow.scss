@import '../../../../styles/variables';

.preset-loader-dialog-row {
  display: flex;
  width: 100%;
}

.preset-loader-dialog-row-label {
  flex: 1;
  line-height: 2.5em;
}
