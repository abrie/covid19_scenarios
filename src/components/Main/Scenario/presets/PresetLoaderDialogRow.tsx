import React from 'react'
import { useTranslation } from 'react-i18next'
import { Button, NavLink } from 'reactstrap'
import './PresetLoaderDialog.scss'

export interface LoadPresetDialogRecordProps {
  label: string
  value: string
}

export interface LoadPresetDialogRowProps {
  data: LoadPresetDialogRecordProps
  onLoadButtonClick: (id: string) => void
}

const PresetLoaderDialogRow = ({ data, onLoadButtonClick }: LoadPresetDialogRowProps) => {
  const { t } = useTranslation()

  return (
    <tr className="preset-loader-dialog-row">
      <td className="preset-loader-dialog-row-label">
        <NavLink
          data-testid="PresetLoaderDialogRowLink"
          onClick={(event) => {
            event.preventDefault()
            onLoadButtonClick(data.value)
          }}
        >
          {data.label}
        </NavLink>
      </td>
      <td>
        <Button
          data-testid="PresetLoaderDialogRowButton"
          onClick={() => {
            onLoadButtonClick(data.value)
          }}
        >
          {t('Load')}
        </Button>
      </td>
    </tr>
  )
}

export default PresetLoaderDialogRow
