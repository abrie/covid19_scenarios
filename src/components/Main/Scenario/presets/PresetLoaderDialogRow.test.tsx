import React from 'react'
import { noop } from 'lodash'
import { cleanup, fireEvent, render, RenderResult } from '@testing-library/react'
import PresetLoaderDialogRow from './PresetLoaderDialogRow'

const DATA = {
  label: 'Some label',
  value: '123',
}

describe('PresetLoaderDialogRow', () => {
  let onLoadButtonClick: (id: string) => void
  let wrapper: RenderResult

  beforeEach(() => {
    console.warn = noop
    console.error = noop

    onLoadButtonClick = jest.fn()
    wrapper = render(<PresetLoaderDialogRow data={DATA} onLoadButtonClick={onLoadButtonClick} />)
  })

  afterEach(cleanup)

  it('matches snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('renders label and button', () => {
    expect(wrapper.getByText('Some label')).not.toBeNull()
    expect(wrapper.getByTestId('PresetLoaderDialogRowLink')).not.toBeNull()
    expect(wrapper.getByTestId('PresetLoaderDialogRowButton')).not.toBeNull()
  })

  it('link click triggers onLoadButtonClick handler', () => {
    const loadButton = wrapper.getByTestId('PresetLoaderDialogRowLink')
    fireEvent.click(loadButton)

    expect(onLoadButtonClick).toHaveBeenCalledWith('123')
  })

  it('button click triggers onLoadButtonClick handler', () => {
    const loadButton = wrapper.getByTestId('PresetLoaderDialogRowButton')
    fireEvent.click(loadButton)

    expect(onLoadButtonClick).toHaveBeenCalledWith('123')
  })
})
