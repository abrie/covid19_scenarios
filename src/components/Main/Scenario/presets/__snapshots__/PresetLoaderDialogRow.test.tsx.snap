// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PresetLoaderDialogRow matches snapshot 1`] = `
<DocumentFragment>
  <tr
    class="preset-loader-dialog-row"
  >
    <td
      class="preset-loader-dialog-row-label"
    >
      <a
        class="nav-link"
        data-testid="PresetLoaderDialogRowLink"
      >
        Some label
      </a>
    </td>
    <td>
      <button
        class="btn btn-secondary"
        data-testid="PresetLoaderDialogRowButton"
        type="button"
      >
        Load
      </button>
    </td>
  </tr>
</DocumentFragment>
`;
